### DISCUSSION TITLE
Difficulty in using LDAP with Minio

### DISCUSSION THREAD

**Discussion by qosyx** [minio/minio#20066] [Q&A]
I'm trying to connect my minio server to an ldap. The connection worked, but I can't map the users. The Minio server replies that it can't find the user

When I search for the user with ldapsearch -x -H ldap://172.17.0.2:1389 -D "cn=admin,dc=minio,dc=com" -w adminpassword -b "dc=minio,dc=com", I find it.
Looking at the ldap logs, I observed the following

668e4340.38e07130 0xffff5d5cd160 conn=1034 op=1 SRCH base="dc=minio,dc=com" scope=2 deref=0 filter="(uid=asmith)"

on the other hand, using minio I get this

668e438d.1b918fdb 0xffff5cabc160 conn=1035 op=1 SRCH base="uid=asmith,cn=users,cn=admin,dc=minio,dc=com" scope=0 deref=0 filter="(objectClass=*)"

Translated with DeepL.com (free version)

**Comment by harshavardhana** [comment]
it looks like you have wrongly configured your LDAP can you provide the LDAP settings?

**Reply by qosyx** [reply]
my ldap conf

docker run --detach --rm --name openldap \
--network minio \
  --env LDAP_ADMIN_USERNAME=admin \
  --env LDAP_ADMIN_PASSWORD=adminpassword \
  --env LDAP_USERS=customuser \
  --env LDAP_PASSWORDS=custompassword \
  --env LDAP_ROOT=dc=minio,dc=com \
  --env LDAP_ADMIN_DN=cn=admin,dc=minio,dc=com \
  -p 1389:1389 \
  bitnami/openldap:latest


user.diff
dn: cn=Alice Smith,ou=users,dc=minio,dc=com
objectClass: inetOrgPerson
cn: Alice Smith
sn: Smith
uid:asmith
givenName: Alice
mail: alicesmith@example.com
userPassword: password123

minio 

docker run -p 9000:9000 -p 9001:9001 \
--network minio \
  quay.io/minio/minio server /data --console-address ":9001"
  
  connect ldap with minio
  
  mc idp ldap add --debug myminio \
  server_addr=172.17.0.2:1389 \
  lookup_bind_dn=cn=admin,dc=minio,dc=com \
  lookup_bind_password=adminpassword \
  user_dn_search_base_dn=dc=minio,dc=com \
  user_dn_search_filter="(uid=%s)" \
  group_search_base_dn=ou=users,dc=minio,dc=io \
  group_search_filter="(&(objectclass=groupofnames)(member=%d))"

**Reply by qosyx** [reply]
any update for me ?

**Comment by qosyx** [comment]
any update for me ?

**Reply by klauspost** [reply]
@qosyx If you want updates, look into our [subscription offer](https://min.io/pricing).
